### PR TITLE
fix(cp): avoid showing empty descriptor in progress bar

### LINKED ISSUE #1442: progress bar missing when copying artifact with zero-length layer from registry to OCI layout
### What happened in your environment?

Tried to copy a self-made artifact with multiple zero-length layers from registry to local OCI image layout but see a progress bar missing in the terminal output
```console
> oras cp --to-oci-layout jinzha1.azurecr.io/test:blank-progress .
✓ Copied  application/vnd.docker.container.image.v1+json                                                              0/0  B 100.00%    9ms
  └─ sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
⠋ [....................](   0  B/s) Copying application/vnd.docker.image.rootfs.diff.tar.gzip                      0.00/0  B   0.00%   26ms
  └─ sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
✓ Copied  application/vnd.docker.distribution.manifest.v2+json                                                  6.16/6.16 kB 100.00%   19ms
  └─ sha256:47506275bd97c02cd5f2c8c86d67d1faf8d3816ba0278aae63fb9cd25ec02e6c
Copied [registry] jinzha1.azurecr.io/test:blank-progress => [oci-layout] .
Digest: sha256:47506275bd97c02cd5f2c8c86d67d1faf8d3816ba0278aae63fb9cd25ec02e6c
```

### What did you expect to happen?

Should show copied for the `application/vnd.docker.image.rootfs.diff.tar.gzip` layer

### How can we reproduce it?

Run ` oras cp --to-oci-layout jinzha1.azurecr.io/test:blank-progress .`

### What is the version of your ORAS CLI?

Version:        1.2.0
Go version:     go1.22.3
Git commit:     dcef719e208a9b226b15bc6512ad729a7dd93270
Git tree state: clean

### What is your OS environment?

Window 11

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by qweeah**
**What this PR does / why we need it**:
This PR fixes progress ratio and missing descriptor in the progress output.

Before
```console
> oras cp --to-oci-layout jinzha1.azurecr.io/test:blank-progress .
✓ Copied  application/vnd.docker.container.image.v1+json                                                              0/0  B 100.00%    9ms
  └─ sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
⠋ [....................](   0  B/s) Copying application/vnd.docker.image.rootfs.diff.tar.gzip                      0.00/0  B   0.00%   26ms
  └─ sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
✓ Copied  application/vnd.docker.distribution.manifest.v2+json                                                  6.16/6.16 kB 100.00%   19ms
  └─ sha256:47506275bd97c02cd5f2c8c86d67d1faf8d3816ba0278aae63fb9cd25ec02e6c
Copied [registry] jinzha1.azurecr.io/test:blank-progress => [oci-layout] .
Digest: sha256:47506275bd97c02cd5f2c8c86d67d1faf8d3816ba0278aae63fb9cd25ec02e6c
```

After
```console
> oras cp --to-oci-layout jinzha1.azurecr.io/test:blank-progress ./debug
✓ Copied  application/vnd.docker.image.rootfs.diff.tar.gzip                                                                                                0/0  B 100.00%    1ms
  └─ sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
✓ Copied  application/vnd.docker.container.image.v1+json                                                                                                   0/0  B 100.00%     0s
  └─ sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
✓ Copied  application/vnd.docker.distribution.manifest.v2+json                                                                                       6.16/6.16 kB 100.00%  808µs
  └─ sha256:47506275bd97c02cd5f2c8c86d67d1faf8d3816ba0278aae63fb9cd25ec02e6c
Copied [registry] jinzha1.azurecr.io/test:blank-progress => [oci-layout] ./debug
Digest: sha256:47506275bd97c02cd5f2c8c86d67d1faf8d3816ba0278aae63fb9cd25ec02e6c
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1442 

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: cmd/oras/internal/display/status/progress/status.go
[the remark this answers is not in the document]
Comment (qweeah) [reply cmd/oras/internal/display/status/progress/status.go:1]: Discussed offline and receiving comment on improving the code readability. Addressed accordingly.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**